### ISSUE TITLE
pip install xrootd not working

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#1377]
```
pip3_install_user xrootd
Collecting xrootd
Using cached xrootd-5.0.3.tar.gz (2.8 MB)
ERROR: Requested xrootd from https://files.pythonhosted.org/packages/b8/29/aba669bb59fd61c29a556e2dd38b921e57c3e14a98edbdff54462248ddf6/xrootd-5.0.3.tar.gz#sha256=e2d600ba821d5ab5c78bbee4473bb6b51cac3efdb9fe41fb49b9afe3d8dc0afe has different version in metadata: '0.0.0'
```

**Comment by simonmichal** [comment]
@adriansev: thanks for reporting this problem!

I tried following pip commands: `pip install xrootd`, `pip install --user xrootd`, `pip3 install xrootd` and `pip3 install --user xrootd` and all work for me ;-)

Could you tell me what the `pip3_install_user` actually is?

**Comment by adriansev** [comment]
@simonmichal 
```
ALIBUILD##[Friday 08.01.21 22:56] adrian@hal : ~  $
alias pip3_install_user
alias pip3_install_user='pip3 install --upgrade --user'
```

**Comment by simonmichal** [comment]
got it! I finally managed to reproduce it!

**Comment by simonmichal** [comment]
this only happens for python3 and only if one tries to install from pip repository, if I download the wheel manually (say with wget) and then install it with pip3 it all works : |

**Comment by simonmichal** [comment]
@adriansev : this commit 68979c3b2e88a85336087c916d13babd5ca69fdd should fix the problem, though we will be sure only once it is released so I'm not closing the ticket.

**Comment by nikoladze** [comment]
To add to this - there also seems to be an issue that some of the installed libraries are not found (if the xrootd libraries are not already installed in the system). I can reproduce this with the `python:3.8-buster` docker image:

```sh
$ docker run --rm -it python:3.8-buster bash
$ apt-get update && apt-get install cmake
$ # in this version there also seems to be an issue with the new dependency resolver ...
$ python3 -m pip --use-deprecated legacy-resolver install xrootd
$ python
```

```pycon
>>> import pyxrootd.client
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: libXrdXml.so.3: cannot open shared object file: No such file or directory
```

```sh
$ ldd /usr/local/lib/python3.8/site-packages/pyxrootd/client.cpython-38-x86_64-linux-gnu.so
        linux-vdso.so.1 (0x00007fff85b5b000)
        libXrdCl.so.3 => /usr/local/lib/python3.8/site-packages/pyxrootd/lib/libXrdCl.so.3 (0x00007fa4112da000)
        libXrdUtils.so.3 => /usr/local/lib/python3.8/site-packages/pyxrootd/lib/libXrdUtils.so.3 (0x00007fa41121e000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa411210000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fa41108c000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa410f09000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa410eef000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa410ecc000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa410d0b000)
        libXrdXml.so.3 => not found
        libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007fa410d02000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fa410ae4000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa410ada000)
        libssl.so.1.1 => /usr/lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007fa410a46000)
        libcrypto.so.1.1 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007fa41075d000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa411465000)
```

The problem may be that `libXrdXml.so.3` is not directly linked, but needed for ` libXrdCl.so.3`

```
$ objdump -p /usr/local/lib/python3.8/site-packages/pyxrootd/client.cpython-38-x86_64-linux-gnu.so  | grep RUNPATH
  RUNPATH              $ORIGIN/lib
$ objdump -p /usr/local/lib/python3.8/site-packages/pyxrootd/client.cpython-38-x86_64-linux-gnu.so | grep NEEDED
  NEEDED               libXrdCl.so.3
  NEEDED               libXrdUtils.so.3
  NEEDED               libdl.so.2
  NEEDED               libstdc++.so.6
  NEEDED               libm.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libpthread.so.0
  NEEDED               libc.so.6
$ objdump -p /usr/local/lib/python3.8/site-packages/pyxrootd/lib/libXrdCl.so.3 | grep NEEDED
  NEEDED               libXrdXml.so.3
  NEEDED               libXrdUtils.so.3
  NEEDED               libpthread.so.0
  NEEDED               libuuid.so.1
  NEEDED               libz.so.1
  NEEDED               librt.so.1
  NEEDED               libdl.so.2
  NEEDED               libssl.so.1.1
  NEEDED               libcrypto.so.1.1
  NEEDED               libstdc++.so.6
  NEEDED               libm.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
  NEEDED               ld-linux-x86-64.so.2
```

**Comment by simonmichal** [comment]
@nikoladze : thanks for reporting this issue and special thanks for the clear instructions how to reproduce the problem, I'll have a look how to fix it

**Comment by simonmichal** [comment]
@adriansev : actually I reverted the original patch and submitted a new one: 222014a65ed28f4ef26bd501e18516a06d3386b0, I tested it with `testpypi` repo and it works fine

@nikoladze : now, when I install a xrootd `HEAD` from `testpypi` on `python:3.8-buster` it also seems that your problem has been fixed

I'll push the next release candidate to `testpypi` for you guys to test and then hopefully we can close this one ;-)

**Comment by nikoladze** [comment]
Thanks! @simonmichal The `HEAD` version you are referring to is that https://test.pypi.org/project/xrootd/7.7.20 ?

With that the global installation in the docker container works. However, the case where i originally found the issue was installing in a virtual environment. I didn't know this would make a difference, but apparently this is still a problem:

```
$ docker run --rm -it python:3.8-buster bash
$ apt-get update && apt-get -y install cmake
$ python3 -m venv test-venv
$ source test-venv/bin/activate
$ pip install wheel
$ pip install --index-url https://test.pypi.org/simple/ xrootd==7.7.20
$ python
>>> import pyxrootd.client
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'pyxrootd.client'
```

The `client.<something>.so` file seems missing:

```
$ ls /test-venv/lib/python3.8/site-packages/pyxrootd/
include  lib  share
```

For comparison when installing globally (not in venv) the folder looks like this:

```
$ ls /usr/local/lib/python3.8/site-packages/pyxrootd
_init__.py  __pycache__  client.cpython-38-x86_64-linux-gnu.so  include  lib  share
```

Also, the `XRootD` module folder is missing when installing in the venv.

It is maybe worth noting that during installation pip tries to install via "wheel" and then falls back to `setup.py` install. In newer versions there is this warning: "DEPRECATION: Could not build wheels for xrootd which do not use PEP 517. pip will fall back to legacy 'setup.py install' for these. pip 21.0 will remove support for this functionality."
I don't really know much about the different ways to make python packages, but maybe this should be kept in mind since it sounds like it could break the package again in the future.

**Comment by simonmichal** [comment]
@nikoladze : thanks for testing the global installation 👍 , I see what the problem is with virtual environment, the `client.cpython-38-x86_64-linux-gnu.so` gets actually installed outside of the environment: 
`/usr/local/lib/python3.8/site-packages/pyxrootd/client.cpython-38-x86_64-linux-gnu.so`

I will create a separate issue where we can handle this problem and close this one, as the original issue have been solved.

**Comment by simonmichal** [comment]
@nikoladze : let's follow up on the virtual env problem in #1397
